pkg/cli/doctor.go

```diff
@@ -166,7 +163,61 @@ func runZipDirDoctor(cmd *cobra.Command, args []string) (retErr error) {
 		}
 		ts := hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
 		descTable = append(descTable, doctor.DescriptorTableRow{ID: int64(i), DescBytes: descBytes, ModTime: ts})
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	namespaceFile, err := os.Open(path.Join(args[0], "system.namespace2.txt"))
+	if err != nil {
+		return err
+	}
+	defer namespaceFile.Close()
+	namespaceTable := make([]doctor.NamespaceTableRow, 0)
+	if err := tableMap(namespaceFile, func(row string) error {
+		fields := strings.Fields(row)
+		parID, err := strconv.Atoi(fields[0])
+		if err != nil {
+			return errors.Errorf("failed to parse parent id %s: %v", fields[0], err)
+		}
+		parSchemaID, err := strconv.Atoi(fields[1])
+		if err != nil {
+			return errors.Errorf("failed to parse parent schema id %s: %v", fields[1], err)
+		}
+		id, err := strconv.Atoi(fields[3])
+		if err != nil {
+			if fields[3] == "NULL" {
+				id = int(descpb.InvalidID)
+			} else {
+				return errors.Errorf("failed to parse id %s: %v", fields[3], err)
+			}
+		}
+
+		namespaceTable = append(namespaceTable, doctor.NamespaceTableRow{
+			NameInfo: descpb.NameInfo{
+				ParentID: descpb.ID(parID), ParentSchemaID: descpb.ID(parSchemaID), Name: fields[2],
+			},
+			ID: int64(id),
+		})
+		return nil
+	}); err != nil {
+		return err
 	}
 
-	return wrapExamine(descTable)
+	return wrapExamine(descTable, namespaceTable)
+}
+
+// tableMap applies `fn` to all rows in `in`.
+func tableMap(in io.Reader, fn func(string) error) error {
```

Comment:
why can't you use go's csv reader or something?

Reply:
thank you - of course!